src/core/thread.d

```diff
@@ -3058,14 +3038,15 @@ do
 * A callback for thread errors in D during collections. Since an allocation is not possible
 *  a preallocated ThreadError will be used as the Error instance
 *
+* Returns:
+*  never returns
 * Throws:
 *  ThreadError.
 */
-private void onThreadError(string msg = null, Throwable next = null) nothrow
+private void onThreadError(string msg) nothrow
 {
     __gshared ThreadError error = new ThreadError(null);
     error.msg = msg;
-    error.next = next;
```

Comment:
While the parameter to onThreadError seems unused, resetting `next` to `null` should remain, as it might have been set in a previously caught exception.

Reply:
good catch

Reply:
Wait a minute, `error` is newly new'd, so it should be `null` allready

Reply:
No, it's `__gshared`, so preallocated in the data segment.

Reply:
The handle is put in __gshared, not the actual object instance.

Reply:
Sure? Try it here: https://run.dlang.io/gist/ae0a51ac41c1e08251d7191cdf67bc13?compiler=dmd

Reply:
yes, you're right, I forgot that it was statically initialized. Thanks for the correction.

Reply:
This leads to a problem with the ref counting. It's getting late, I'll try to fix that tomorrow.

Reply:
fixed